src/main/java/com/internhub/data/positions/scrapers/candidates/DepthAwareCandidate.java

```diff
@@ -0,0 +1,14 @@
+package com.internhub.data.positions.scrapers.candidates;
+
+public class DepthAwareCandidate extends QueuedCandidate {
```

Comment:
I don't agree with making this class. The only reason you would do something like this is if you had plans to switch from a BFS to some other kind of search for the project, and I don't think any other kind of search is possible. Right now this is adding unnecessary complexity and you can just add a depth field to the QueuedCandidate. I don't even see why it's necessary to have a depth field, you can just maintain the depth in the BFS itself, why should Candidates have to worry about at what depth they were added?